supreme/src/iosMain/kotlin/at/asitplus/signum/supreme/crypt/Encryptor.ios.kt

```diff
@@ -0,0 +1,129 @@
+        is SymmetricEncryptionAlgorithm.AES.CBC.Plain -> {
+            return catching {
+                val padded = (alg as AES<*>).addPKCS7Padding(data)
+                val bytes: ByteArray = swiftcall {
+                    CBC.crypt(kCCEncrypt.toLong(), padded.toNSData(), platformData.toNSData(), nsIV, error)
+                }.toByteArray()
+                Ciphertext.Unauthenticated(
+                    alg as SymmetricEncryptionAlgorithm.Unauthenticated,
+                    bytes,
+                    iv
+                ) as Ciphertext<out A, SymmetricEncryptionAlgorithm<A>>
+            }
+        }
+
+        is SymmetricEncryptionAlgorithm.AES.GCM -> {
+            return catching {
+
+                require(iv != null) { "AES implementation error, please report this bug" }
+                val ciphertext = GCM.encrypt(data.toNSData(), platformData.toNSData(), nsIV, nsAAD)
+                if (ciphertext == null) throw UnsupportedOperationException("Error from swift code!")
+
+                return@catching Ciphertext.Authenticated(
+                    alg as SymmetricEncryptionAlgorithm.Authenticated,
+                    ciphertext.ciphertext().toByteArray(),
+                    ciphertext.iv().toByteArray(),
+                    ciphertext.authTag().toByteArray(),
+                    aad
+
+                ) as Ciphertext<A, SymmetricEncryptionAlgorithm<A>>
+            }
+        }
+
+        else -> TODO()
+    }
+}
+
+private fun BlockCipher<*>.addPKCS7Padding(plain: ByteArray): ByteArray {
+    val blockBytes = blockSizeBits.toInt() / 8
+    val diff = blockBytes - (plain.size % blockBytes)
+    return if (diff == 0)
+        plain + ByteArray(blockBytes) { blockBytes.toByte() }
+    else plain + ByteArray(diff) { diff.toByte() }
+}
+
+
+private fun BlockCipher<*>.removePKCS7Padding(plainWithPadding: ByteArray): ByteArray {
+    val paddingBytes = plainWithPadding.last().toInt()
+    require(paddingBytes > 0) { "Illegal padding: $paddingBytes" }
+    require(plainWithPadding.takeLast(paddingBytes).all { it.toInt() == paddingBytes }) { "Padding not consistent" }
+    require(plainWithPadding.size - paddingBytes >= 0) { "Too much padding: data ${plainWithPadding.joinToString()}" }
+    return plainWithPadding.sliceArray(0..<plainWithPadding.size - paddingBytes)
+}
+
+
+@OptIn(ExperimentalForeignApi::class)
+actual internal fun Ciphertext.Authenticated.doDecrypt(secretKey: ByteArray): KmmResult<ByteArray> {
+    return catching {
+        if (algorithm !is SymmetricEncryptionAlgorithm.WithIV<*>) TODO()
+        require(iv != null) { "IV must not be null!" }
+        require(algorithm is AES<*>) { "Only AES is supported" }
+        swiftcall {
+            GCM.decrypt(
+                encryptedData.toNSData(),
+                secretKey.toNSData(),
+                iv!!.toNSData(),
+                authTag.toNSData(),
+                aad?.toNSData(),
+                error
+            )!!.toByteArray()
```

Comment:
`!!` is not necessary

Reply:
compiler begs to differ

Reply:
move `toByteArray` outside `swiftcall`, the nonnull assertion will break swift exception forwarding